connect/api/src/main/java/org/apache/kafka/connect/data/ConnectSchema.java

```diff
@@ -283,20 +285,20 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
         ConnectSchema schema = (ConnectSchema) o;
         return Objects.equals(optional, schema.optional) &&
+                Objects.equals(version, schema.version) &&
```

Comment:
Are these 3 lines an artifact of code reformatting? They don't seem to correspond to changes other than the reordering itself. The previous order matches the variable declaration order, so maybe we want to leave this as-is.

Reply:
Are you implying cache efficiency? 

I'd imagine the order to matter if one of the fields had more chances to return `false` than the others. But I don't think it's worth thinking about this here. Feel free to leave this as you have it. When I saw it, I just thought your editor performed the reordering.

Reply:
@kkonstantine I think the point is to avoid walking nested data structures that would require pointer chasing, so essentially yes it is a cache efficiency thing. If we really wanted to optimize this we would want to make sure everything that can be inlined in the class is checked first, i.e. `optional`, `type`, and `version`, before other simple fields like `name` and `doc`.